tests/Controller/TaxonApiTest.php

```diff
@@ -0,0 +1,434 @@
+        $this->assertResponse($response, 'taxon/create_root_with_multiple_translations_response', Response::HTTP_CREATED);
+    }
+
+    /**
+     * @test
+     */
+    public function it_allows_create_root_taxon()
+    {
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+
+        $data =
+<<<EOT
+        {
+            "code": "fluffy_pets"
+        }
+EOT;
+
+        $this->client->request('POST', '/api/v1/taxons/', [], [], static::$authorizedHeaderWithContentType, $data);
+
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'taxon/create_root_response', Response::HTTP_CREATED);
+    }
+
+    /**
+     * @test
+     */
+    public function it_allows_create_taxon_with_multiple_translations()
+    {
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+        $this->loadFixturesFromFile('resources/locales.yml');
+        $this->loadFixturesFromFile('resources/taxons.yml');
+
+        $data =
+<<<EOT
+        {
+            "code": "fluffy_pets",
+            "parent": "category",
+            "translations": {
+                "en_US": {
+                    "name": "Fluffy Pets",
+                    "slug": "fluffy-pets"
+                },
+                "nl_NL": {
+                    "name": "Pluizige Huisdieren",
+                    "slug": "pluizige-huisdieren"
+                }
+            }
+        }
+EOT;
+
+        $this->client->request('POST', '/api/v1/taxons/', [], [], static::$authorizedHeaderWithContentType, $data);
+
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'taxon/create_with_multiple_translations_response', Response::HTTP_CREATED);
+    }
+
+
+    /**
+     * @test
+     */
+    public function it_allows_create_taxon_with_parent()
+    {
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+        $this->loadFixturesFromFile('resources/taxons.yml');
+
+        $data =
+<<<EOT
+        {
+            "code": "horror",
+            "parent": "books"
+        }
+EOT;
+
+        $this->client->request('POST', '/api/v1/taxons/', [], [], static::$authorizedHeaderWithContentType, $data);
+
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'taxon/create_with_parent_response', Response::HTTP_CREATED);
+    }
+
+    /**
+     * @test
+     */
+    public function it_does_not_allow_to_create_taxon_without_required_fields()
+    {
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+
+        $this->client->request('POST', '/api/v1/taxons/', [], [], static::$authorizedHeaderWithContentType);
+
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'taxon/create_validation_fail_response', Response::HTTP_BAD_REQUEST);
+    }
+
+    /**
+     * @test
+     */
+    public function it_allows_create_taxon_with_images()
```

Comment:
Here and everywhere where it's needed, we should use `it_allows_to_...` or `it_allows_...ing_...`